capstone/src/main/webapp/script.js

```diff
@@ -1,31 +1,153 @@
+  const zipControlDiv = document.createElement('div');
+  const zipControl = new ZipControl(zipControlDiv, map);
+  map.controls[google.maps.ControlPosition.LEFT_CENTER].push(zipControlDiv);
+
+  //* *button for zipcode data layer */
+  const precinctControlDiv = document.createElement('div');
+  const precinctControl = new PrecinctControl(precinctControlDiv, map);
+  map.controls[google.maps.ControlPosition.LEFT_CENTER]
+      .push(precinctControlDiv);
+}
+
+function CenterControl(controlDiv, map) {
+  //* *button creation and positioning*/
+  const controlUI = document.createElement('div');
+  controlUI.style.backgroundColor = '#fff';
+  controlUI.style.border = '2px solid #fff';
+  controlUI.style.borderRadius = '3px';
+  controlUI.style.boxShadow = '0 2px 6px rgba(0,0,0,.3)';
+  controlUI.style.cursor = 'pointer';
+  controlUI.style.marginBottom = '22px';
+  controlUI.style.textAlign = 'center';
+  controlUI.title = 'Click to recenter the map';
+  controlDiv.appendChild(controlUI);
+
+  //* *css for interior of all buttons*/
+  const text = document.createElement('div');
+  text.style.color = 'rgb(25,25,25)';
+  text.style.fontFamily = 'Roboto,Arial,sans-serif';
+  text.style.fontSize = '16px';
+  text.style.lineHeight = '38px';
+  text.style.paddingLeft = '5px';
+  text.style.paddingRight = '5px';
+  text.innerHTML = 'Center Map';
+  controlUI.appendChild(text);
+
+  //* *button functionality */
+  controlUI.addEventListener('click', function() {
+    map.setCenter({lat: 37.7749, lng: -122.4194});
+  });
+}
+
+let zipClicks = 0;
+function ZipControl(controlDiv, map) {
+  //* *adding zipcode overlay*/
+  const zipcodeLayer = new google.maps.Data({map: map});
+  zipcodeLayer.loadGeoJson('zipcode-data.json');
+  zipcodeLayer.setStyle({visible: false});
+
+  //* *button creation and positioning*/
+  const controlUI = document.createElement('div');
+  controlUI.style.backgroundColor = '#fff';
+  controlUI.style.border = '2px solid #fff';
+  controlUI.style.borderRadius = '3px';
+  controlUI.style.boxShadow = '0 2px 6px rgba(0,0,0,.3)';
+  controlUI.style.cursor = 'pointer';
+  controlUI.style.marginBottom = '22px';
+  controlUI.style.textAlign = 'center';
+  controlUI.title = 'Click to show San Fransisco zip codes';
+  controlDiv.appendChild(controlUI);
+
+  //* *css for interior of all buttons*/
+  const text = document.createElement('div');
+  text.style.color = 'rgb(25,25,25)';
+  text.style.fontFamily = 'Roboto,Arial,sans-serif';
+  text.style.fontSize = '16px';
+  text.style.lineHeight = '38px';
+  text.style.paddingLeft = '5px';
+  text.style.paddingRight = '5px';
+  text.innerHTML = 'Show Zipcodes';
+  controlUI.appendChild(text);
+
+  //* *button functionality */
+  controlUI.addEventListener('click', function() {
+    zipClicks++;
+    if (zipClicks%2==1) {
+      zipcodeLayer.setStyle({visible: true});
+    } else {
+      zipcodeLayer.setStyle({visible: false});
+    }
+  });
+}
+
+let precinctClicks = 0;
```

Comment:
Same as above with zipClicks. Also if you have a variable that's called precinctDisplayOn its a lot clearer what the purpose is